app/src/main/java/com/mobven/shortly/ui/list/MyLinksFragment.kt

```diff
@@ -67,36 +71,43 @@ class MyLinksFragment : Fragment() {
             )
         }
 
-        shortLinkPagingAdapter.itemClickListener = {
-            viewModel.selectedShortenData(true, it.code)
-            shortLinkPagingAdapter.copiedItem = it.code
-            val clip = ClipData.newPlainText(getString(R.string.copied), it.short_link)
-            clipBoardManager.setPrimaryClip(clip)
-        }
-
-        shortLinkPagingAdapter.itemShareListener = {
-            requireContext().share(it.short_link, getString(R.string.share))
-        }
+            shortLinkPagingAdapter.itemClickListener = { shortenData, position ->
+                analyticsManagerImpl.copyClickEvent(position)
+                viewModel.selectedShortenData(true, shortenData.code)
+                shortLinkPagingAdapter.copiedItem = shortenData.code
+                val clip = ClipData.newPlainText(getString(R.string.copied), shortenData.short_link)
+                clipBoardManager.setPrimaryClip(clip)
+            }
 
-        shortLinkPagingAdapter.itemRemoveListener = { code, shortLink ->
-            if (clipBoardManager.primaryClip?.getItemAt(0)?.text?.toString()
-                    .equals(shortLink) && (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P)
-            ) {
-                clipBoardManager.clearPrimaryClip()
+            shortLinkPagingAdapter.itemShareListener = { item, position ->
+                analyticsManagerImpl.shareClickEvent(position)
+                requireContext().share(item.short_link, getString(R.string.share))
             }
-            viewModel.delete(code)
-        }
 
-        shortLinkPagingAdapter.openUrl = {
-            val browserIntent = Intent(Intent.ACTION_VIEW, Uri.parse(it))
-            startActivity(browserIntent)
+            shortLinkPagingAdapter.itemRemoveListener = { code, shortLink, position ->
+                if (clipBoardManager.primaryClip?.getItemAt(0)?.text?.toString()
+                        .equals(shortLink) && (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P)
+                ) {
+                    clipBoardManager.clearPrimaryClip()
+                }
+                analyticsManagerImpl.deleteClickEvent(position)
+                viewModel.delete(code)
+
+
+                shortLinkPagingAdapter.openUrl = {
+                val browserIntent = Intent(Intent.ACTION_VIEW, Uri.parse(it))
+                startActivity(browserIntent)
```

Comment:
Bu şekilde daha kısa olur sanki değişkenden de kurtuluruz. Bir de reformat rica edeyim.
Intent(Intent.ACTION_VIEW, Uri.parse(it)).apply { startActivity(this) }